OVERVIEW.md

```diff
@@ -0,0 +1,71 @@
+
+# Object
+
+All bauble source files consists of a set of objets. A single Bauble object is a tree of Bauble values and a type. A Bauble object can be thought of as a single asset, being tied to a unique path. Objects are the format of its contents Bauble provides after it has parsed all of its files, and ultimately what is used to convery the parsed Bauble contents. 
+
+# Values
+
+ A value in Bauble is typed, and contains an enum representing the kind of the value (whether it is a number, a string, an array or a map for example). Values are not that usually interacted with by the end user, as generally Bauble is implemented through the derive macro which means the user does not need to handle parsing from a raw Bauble value themselves.
+
+# Types
+
+Types are registered to the [`BaubleContext`] through the builder, and every type requires the [`Bauble`] trait to be implemented on top of it. The [`Bauble`] trait determines how the type gets registered into the [`BaubleContext`], how it is parsed from Bauble source, and what path the type has. In Bauble every object is associated with a type, the type may be explicitly written with the value or implied by context, similar to type inference rules in Rust.
+
+# Traits
+
+Traits may be registered to Bauble in the form of `dyn Trait` types. Every type registered to Bauble has to be explicitly set to derive a specific trait, Bauble is unable to use reflection and figure out if a type implements a trait based purely on if it implements that trait in Rust, which is an unfortunate limitation. If a trait has been registered, `dyn Trait` can then be used within registered Bauble types. In order for Bauble to parse a value of type `dyn Trait` with a value of a type that implements `Trait` called `T`, both `Trait` the `T` must be registered, and Bauble must know `T` implements `Trait`.
+
+# Modules
+
+Every registered source file in Bauble is a module, similar to Rust. Every module contains various assets. The notion of sub-modules are not really present in Bauble.
+
+# Paths
+
+Every asset, module and reference in Bauble has corresponding unique path. In Bauble this is known as [`TypePath`], and is the association to that particular element in the [`BaubleContext`]. A path consists of various elements. Similar to Rust, most elements are seperated by `::`, so `a::b` means element `b` which is a child of element `a`. An element here can be a module, type or object. There are things known as sub-objects which may be appended to the path of a regular object, which are effectively children of the current object. A sub-object's path is denoted by `<path to parent>@$idx` where `$idx` is the index of the subobject to the parent (the first subject being 0, the second 1, the third 2, etc).
```

Comment:
>  which are effectively children of the current object

although they are represented as siblings in the current module?

Reply:
Right, what I'm saying is that most of the code treats `object@x` as an independent object in `root::mod` (ie as sibling to `object`)

Reply:
I see what you mean. This is true, although personally I would consider their direct parent `object` and not `mod` I do see this perspective too.